### ISSUE TITLE
Firehose returns 500 on access to non existing stream

### ISSUE THREAD

**Issue by yaitskov** [localstack/localstack#2879]
┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-238) by [Unito](https://www.unito.io/learn-more)

**Comment by usmangani1** [comment]
@yaitskov Thanks for reporting the issue.
Can you please help us with a sample script or a command using which you are trying to access the non existing stream?

**Comment by usmangani1** [comment]
@yaitskov  Can you help us here ?

**Comment by tekener** [comment]
Same here, i got a 500 on a not existing stream with localstack 12.1

```
2020-11-03T08:24:58:ERROR:firehose_api: Exception on / [POST]

Traceback (most recent call last):

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app

response = self.full_dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request

rv = self.handle_user_exception(e)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function

return cors_after_request(app.make_response(f(*args, **kwargs)))

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception

reraise(exc_type, exc_value, tb)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise

raise value

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request

rv = self.dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request

return self.view_functions[rule.endpoint](**req.view_args)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 252, in post_request

put_record(stream_name, record)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 55, in put_record

return put_records(stream_name, [record])

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 60, in put_records

for dest in stream['Destinations']:

TypeError: 'NoneType' object is not subscriptable

2020-11-03T08:24:58:ERROR:firehose_api: Exception on / [POST]

Traceback (most recent call last):

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app

response = self.full_dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request

rv = self.handle_user_exception(e)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function

return cors_after_request(app.make_response(f(*args, **kwargs)))

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception

reraise(exc_type, exc_value, tb)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise

raise value

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request

rv = self.dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request

return self.view_functions[rule.endpoint](**req.view_args)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 252, in post_request

put_record(stream_name, record)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 55, in put_record

return put_records(stream_name, [record])

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 60, in put_records

for dest in stream['Destinations']:

TypeError: 'NoneType' object is not subscriptable

2020-11-03T08:24:58:ERROR:firehose_api: Exception on / [POST]

Traceback (most recent call last):

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app

response = self.full_dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request

rv = self.handle_user_exception(e)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function

return cors_after_request(app.make_response(f(*args, **kwargs)))

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception

reraise(exc_type, exc_value, tb)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise

raise value

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request

rv = self.dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request

return self.view_functions[rule.endpoint](**req.view_args)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 252, in post_request

put_record(stream_name, record)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 55, in put_record

return put_records(stream_name, [record])

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 60, in put_records

for dest in stream['Destinations']:

TypeError: 'NoneType' object is not subscriptable

2020-11-03T08:24:58:ERROR:firehose_api: Exception on / [POST]

Traceback (most recent call last):

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app

response = self.full_dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request

rv = self.handle_user_exception(e)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function

return cors_after_request(app.make_response(f(*args, **kwargs)))

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception

reraise(exc_type, exc_value, tb)

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise

raise value

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request

rv = self.dispatch_request()

File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request

return self.view_functions[rule.endpoint](**req.view_args)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 252, in post_request

put_record(stream_name, record)

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 55, in put_record

return put_records(stream_name, [record])

File "/opt/code/localstack/localstack/services/firehose/firehose_api.py", line 60, in put_records

for dest in stream['Destinations']:

TypeError: 'NoneType' object is not subscriptable
```

**Comment by whummer** [comment]
Thanks for reporting @yaitskov @tekener - this should be fixed in c18d7454e4df15d8110d29fdead38220c85b068e . Can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
